core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpDockerCmdExecFactory.java

```diff
@@ -40,6 +40,13 @@ public void init(DockerClientConfig dockerClientConfig) {
         super.init(dockerClientConfig);
 
         OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder()
+            .hostnameVerifier(new HostnameVerifier() {
```

Comment:
I wonder if we should have it as a configuration property (see `TestcontainersConfiguration`)

/cc @rnorth @kiview

Reply:
Also, does it let you use a lambda here?

Reply:
> I wonder if we should have it as a configuration property (see `TestcontainersConfiguration`)
> 
> /cc @rnorth @kiview

totally, I just PR the code to resolve my issue but it can be improved.



> Also, does it let you use a lambda here?
Yes, sorry :
            .hostnameVerifier((hostname, sslSession) -> true)

Reply:
> I wonder if we should have it as a configuration property (see TestcontainersConfiguration)

I'd agree. 

I can understand that it needs to be done under certain circumstances, but I'd prefer to keep the default as-is. Disabling TLS cert verification feels like very much a last resort.

Reply:
> > I wonder if we should have it as a configuration property (see TestcontainersConfiguration)
> 
> I'd agree.
> 
> I can understand that it needs to be done under certain circumstances, but I'd prefer to keep the default as-is. Disabling TLS cert verification feels like very much a last resort.

See my issue #1573: the default configuration for docker-java and docker-client don't check the host in the certificate. I think it would be better to have the same behavior or to access it simply (config maybe). By the way, out-of-the-box, I can't use testcontainers in my environment.

Reply:
I'm fine with adding a property for this, I'm strongly against relaxing our default.
 >  the default configuration for docker-java and docker-client don't check the host in the certificate

Just because they are using insecure defaults doesn't mean we have to follow.
Btw. does your Docker CLI work correctly? Wondering if the Docker client checks.

Reply:
No pb, I'll be fine if I can disable the check in the configuration. And I'm probably not the only one having this issue, maybe it'll be usefull for other users.

FYI, Docker CLI works perfectly. It doesn't care about the unckeched host in certificate, although I set the --tlsverify and cert/key options.